applications/ExaquteSandboxApplication/custom_processes/metrics_divergencefree_process.cpp

```diff
@@ -0,0 +1,305 @@
+                    divergencefree_elem_value = i_neighbour_elements->GetValue(r_reference_var);
+                }
+            }
+
+            // Set element size
+            double element_size;
+            if (divergencefree_elem_value >= mMaxStrategyTargetRefinementCoefficient*mDivergenceFreeMaxValue) {
+                element_size = nodal_h/mMaxStrategyRefinementCoefficient;
+            }
+            else {
+                element_size = nodal_h;
+            }
+
+            // Check with max and min size
+            if (element_size < mMinSize) element_size = mMinSize;
+            if (element_size > mMaxSize) element_size = mMaxSize;
+
+            // Set metric
+            BoundedMatrix<double, TDim, TDim> metric_matrix = ZeroMatrix(TDim, TDim);
+            for(IndexType i = 0;i < TDim; ++i)
+                metric_matrix(i,i) = 1.0/std::pow(element_size, 2);
+
+            // Transform metric matrix to a vector
+            const TensorArrayType metric = MathUtils<double>::StressTensorToVector<MatrixType, TensorArrayType>(metric_matrix);
+
+            // Setting value
+            it_node->SetValue(tensor_variable, metric);
+
+            KRATOS_INFO_IF("MetricDivergenceFreeProcess", mEchoLevel > 2) << "Node " << it_node->Id() << " has metric: "<< metric << std::endl;
+
+        }
+    }
+
+    // Mean distribution strategy
+    else if (mRefinementStrategy == "mean_distribution_strategy") {
+
+        // Reference variable
+        const auto& r_reference_var  = KratosComponents<Variable<double>>::Get(mMeanStrategyReferenceVariable);
+        const auto& r_reference_norm = KratosComponents<Variable<double>>::Get(mMeanStrategyReferenceNorm);
+
+        #pragma omp parallel for
+        for(int i_node = 0; i_node < number_nodes; ++i_node) {
+            auto it_node = nodes_array.begin() + i_node;
+
+            // Initialize Divergence
+            double divergencefree_interp_value = 0;
+            double local_volume = 0;
+
+            // Get divergence interpolation value on the node
+            auto& neigh_elements = it_node->GetValue(NEIGHBOUR_ELEMENTS);
+            for(auto i_neighbour_elements = neigh_elements.begin(); i_neighbour_elements != neigh_elements.end(); i_neighbour_elements++) {
+                divergencefree_interp_value += i_neighbour_elements->GetValue(r_reference_var)*i_neighbour_elements->GetGeometry().Area();
+                local_volume += i_neighbour_elements->GetGeometry().Area();
+            }
+            divergencefree_interp_value /= local_volume;
+
+            // Estimate element size
+            const double nodal_h = it_node->GetValue(NODAL_H);
+
+            // Set element size
+            double element_size;
+            double factor = mMeanStrategyTargetRefinementCoefficient*mMeanStrategyDivergenceFreeOverAllDomain/std::sqrt(number_elements)/divergencefree_interp_value;
+
+            if (factor < 1.0/mMeanStrategyRefinementBound) factor = 1.0/mMeanStrategyRefinementBound;
```

Comment:
Check for non zero values

Reply:
What should I check? `divergencefree_interp_value`? Because `mMeanStrategyRefinementBound` is loaded from the parameters

Reply:
Done